virttest/qemu_devices/qcontainer.py

```diff
@@ -1013,6 +1013,83 @@ def machine_by_params(self, params=None):
             pci_bridge_type = 'pci-bridge'
         pcie_root_port_params = params.get('pcie_root_port_params')
 
+        def pflash_handler(firmware_name, cmd):
+            """
+            Add pflash firmware for EFI boot
+            """
+            machine_cmd = cmd
+            devs = []
+            firmware_path = params.get(firmware_name + "_path")
+            if firmware_path:
+                if not os.path.exists(firmware_path):
+                    raise exceptions.TestError("The firmware path is not exist."
+                                               " Maybe you need to install "
+                                               "related packages.")
+                current_data_dir = params.get("images_base_dir",
+                                              data_dir.get_data_dir())
+                pflash_code_filename = params[firmware_name + "_code_filename"]
+                pflash_code_path = os.path.join(firmware_path,
+                                                pflash_code_filename)
+                pflash_vars_filename = params[firmware_name + "_vars_filename"]
+                pflash_vars_src_path = os.path.join(firmware_path,
+                                                    pflash_vars_filename)
+
+                # To ignore the influence from backends
+                first_image = params.objects('images')[0]
+                img_params = params.object_params(first_image)
+                path = storage.get_image_filename_filesytem(img_params,
+                                                            current_data_dir)
+                img_info = json.loads(qemu_storage.QemuImg(img_params,
+                                                           data_dir.DATA_DIR,
+                                                           path).info(True, "json"))
```

Comment:
I am confused about the way the paths are handled here. In my setup for instance I use "images_base_dir" to define a shared location of all images. In the code above we will possibly check for it and set the `current_data_dir` variable to it. But then `current_data_dir` is not passed to the `QemuImg` as `QemuImg(img_params, current_data_dir)` but through another variable `path` which seems redundant to me and is not used in https://github.com/avocado-framework/avocado-vt/blob/de0996188033f2f9143c4c7df1d9920d50514b6a/virttest/storage.py#L767
thus disregarding the parameter `current_data_dir`. It also seems redundant. How about
```diff
--- a/virttest/qemu_devices/qcontainer.py
+++ b/virttest/qemu_devices/qcontainer.py
@@ -1040,7 +1040,7 @@ class DevContainer(object):
                 path = storage.get_image_filename_filesytem(img_params,
                                                             current_data_dir)
                 img_info = json.loads(qemu_storage.QemuImg(img_params,
-                                                           data_dir.DATA_DIR,
+                                                           current_data_dir,
                                                            path).info(True, "json"))
                 img_name = os.path.basename(img_info.get("full-backing-filename")
                                             or img_info.get("filename"))
```

Reply:
Okay, I got this change. When I submitted this patch, I'm not aware of the introduction of this usage. Will update it.